clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy

```diff
@@ -154,6 +155,28 @@ public class AzureComputeClient extends AzureBaseClient {
     null
   }
 
+  ServiceResponse<Void> destroyServerGroup(String resourceGroupName, String serverGroupName) {
+    ServiceResponse<Void> result = null
+    // The API call might return a timout exception or some other Azure CloudException that is not
+    //   related to the operation we are trying to execute; retry couple of times and if the final
+    //   retry fails then rethrow
+    long operationRetry = 0
+    while (operationRetry < AZURE_ATOMICOPERATION_RETRY) {
+      try {
+        operationRetry ++
+        result = this.client.getVirtualMachineScaleSetsOperations()?.delete(resourceGroupName, serverGroupName)
+        operationRetry = AZURE_ATOMICOPERATION_RETRY
+      } catch (Exception e) {
```

Comment:
I checked the Azure JSDK implementation and for post or delete operations they don't handle 404:
com.microsoft.azure/azure-client-runtime/1.0.0-beta1/79374d64473fc93e2a2478e7b374dd5d289c98b1/azure-client-runtime-1.0.0-beta1-sources.jar!/com/microsoft/azure/AzureClient.java:270
        if (statusCode != 200 && statusCode != 202 && statusCode != 204) {
            CloudException exception = new CloudException(statusCode + " is not a valid polling status code");
            exception.setResponse(response);